src/commands/delete_command.hpp

```diff
@@ -0,0 +1,106 @@
+#ifndef DELETE_COMMAND_HPP
+#define DELETE_COMMAND_HPP
+
+#include "interactive_command_base.hpp"
+#include "command_manager.hpp"
+#include "../core/rq/RegionQueryService.hpp"
+#include "direct_command_base.hpp"
+
+class dicmdDeleteObj : public DirectCommandBase
+{
+public:
+	dicmdDeleteObj(IObjectPoolPtr ptr, QPoint pos)
+		: m_workingSet(std::dynamic_pointer_cast<WorkingSet>(ptr))
+	{
+		add_option("-point",new PointCommandOptionValue(pos));
+	}
+
+	dicmdDeleteObj(IObjectPoolPtr ptr)
+		: m_workingSet(std::dynamic_pointer_cast<WorkingSet>(ptr))
+	{
+		add_option("-point",new PointCommandOptionValue());
+	}
+
+	virtual std::string get_name() override
+	{
+		return "dicmdDeleteShape";
+	}
+
+	virtual void execute() override
+	{
+		// find object from RQ
+		//FIXME need only one init during shape creation
+		RegionQuery& rq = RegionQuery::getInstance();
+		for (auto obj : m_workingSet->getObjects())
+			rq.insertObject(obj);
+
+		QPoint pos = GET_CMD_ARG(PointCommandOptionValue,"-point");
+		IShape* shape = rq.getShapeUnderPos(pos);
+		rq.clear(); // temp solution for the crash
+		if (shape != nullptr)
+		{
+			m_workingSet->deleteShape(shape);
+		}
+
+	}
+
+private:
+	WorkingSetPtr m_workingSet = nullptr;
+};
+
+class InteractiveDeleteAction : public InteractiveCommandBase
+{
+public:
+	InteractiveDeleteAction(IObjectPoolPtr ptr)
+		: m_workingSet(std::dynamic_pointer_cast<WorkingSet>(ptr))
+	{}
+
+	virtual std::string get_name() override
+	{
+		return "incmdDeleteShape";
+	}
+
+	virtual void handle_mouse_click(int x, int y) override
```

Comment:
Actually we don't need it. To grab last mouse position there is 
QPoint InteractiveCommandBase::get_last_point()